geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRangeByScoreExecutor.java

```diff
@@ -0,0 +1,119 @@
+
+    try {
+      byte[] startBytes = commandElements.get(2);
+      byte[] endBytes = commandElements.get(3);
+      if (isNaN(startBytes) || isNaN(endBytes)) {
+        return RedisResponse.error(ERROR_MIN_MAX_NOT_A_FLOAT);
+      }
+      rangeOptions = new SortedSetRangeOptions(startBytes, endBytes);
+    } catch (NumberFormatException ex) {
+      return RedisResponse.error(ERROR_MIN_MAX_NOT_A_FLOAT);
+    }
+
+    // Native redis allows multiple "withscores" and "limit ? ?" clauses; the last "limit"
+    // clause overrides any previous ones
+    if (commandElements.size() >= 5) {
+      int currentCommandElement = 4;
+
+      while (currentCommandElement < commandElements.size()) {
+        try {
+          if (equalsIgnoreCaseBytes(commandElements.get(currentCommandElement),
+              bWITHSCORES)) {
+            withScores = true;
+            currentCommandElement++;
+          } else {
+            parseLimitArguments(rangeOptions, commandElements, currentCommandElement);
+            currentCommandElement += 3;
+          }
+        } catch (NumberFormatException nfex) {
+          return RedisResponse.error(ERROR_NOT_INTEGER);
+        } catch (IllegalArgumentException iex) {
+          return RedisResponse.error(ERROR_SYNTAX);
+        }
+      }
+    }
+
+    // If the range is empty (min == max and both are exclusive,
+    // or limit specified but count is zero), return early
+    if ((rangeOptions.hasLimit() && (rangeOptions.getCount() == 0 || rangeOptions.getOffset() < 0))
+        || (rangeOptions.getStartDouble() == rangeOptions.getEndDouble())
+            && rangeOptions.isStartExclusive() && rangeOptions.isEndExclusive()) {
+      return RedisResponse.emptyArray();
+    }
+    // For ZRANGEBYSCORE, min and max are reversed in order; check if limits are impossible
+    if (isRev() ? (rangeOptions.getStartDouble() < rangeOptions.getEndDouble())
+        : (rangeOptions.getStartDouble() > rangeOptions.getEndDouble())) {
+      return RedisResponse.emptyArray();
+    }
+
+    List<byte[]> result;
+    if (isRev()) {
+      result = redisSortedSetCommands.zrevrangebyscore(command.getKey(), rangeOptions, withScores);
+    } else {
+      result = redisSortedSetCommands.zrangebyscore(command.getKey(), rangeOptions, withScores);
+    }
+
+    return RedisResponse.array(result);
+  }
+
+  void parseLimitArguments(SortedSetRangeOptions rangeOptions, List<byte[]> commandElements,
+      int commandIndex) {
+    int offset;
+    int count;
+    if (equalsIgnoreCaseBytes(commandElements.get(commandIndex), bLIMIT)
+        && commandElements.size() > commandIndex + 2) {
+      offset = narrowLongToInt(bytesToLong(commandElements.get(commandIndex + 1)));
+      count = narrowLongToInt(bytesToLong(commandElements.get(commandIndex + 2)));
+      if (count < 0) {
+        count = Integer.MAX_VALUE;
+      }
+    } else {
+      throw new IllegalArgumentException();
+    }
+    rangeOptions.setLimitValues(offset, count);
```

Comment:
shouldn't we only call setLimitValues if we found the bLIMIT keyword? I guess it works that way (since if we don't find the keyword we throw an exception).  I think the code would be a little cleaner if you declared offset and count inside the if block as you initialize them and made this call at the end of the if block

Reply:
Massaged.